applications/sdp/mspi/src/main.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "hrt/hrt.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/ipc/ipc_service.h>
+#include <zephyr/drivers/mspi.h>
+#include <drivers/mspi/nrfe_mspi.h>
+#include <string.h>
+
+static struct ipc_ept ep;
+static volatile uint32_t bound_sem;
+
+void process_packet(const void *data, size_t len);
+
+static void ep_bound(void *priv)
+{
+	bound_sem = 1;
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	(void)priv;
+
+	process_packet(data, len);
+}
+
+static struct ipc_ept_cfg ep_cfg = {
+	.cb = {
+		.bound = ep_bound,
+		.received = ep_recv,
+	},
+};
+
+void process_packet(const void *data, size_t len)
+{
+	(void)len;
+	nrfe_mspi_flpr_response_t response;
+	uint8_t *buffer = (uint8_t *)data;
+	uint8_t opcode = *buffer++;
```

Comment:
@jaz1-nordic have you changed it? It looks still the same.

Reply:
There is no more buffer-to-buffer copying. One exception is `opcode` reading, because it is used in many places. The rest is casting the pointer to a pointer of another type `uint8_t`, so that you can cast to a struct with a single byte precision. Wasn't that your suggestion?